Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/MessageBuilder.hs

```diff
@@ -155,3 +157,18 @@ buildCollectCashMessage merchantId req = do
   return $
     merchantMessage.message
       & T.replace (templateText "amount") req.amount
+
+data BuildGenericMessageReq = BuildGenericMessageReq {}
+  deriving (Generic)
+
+buildGenericMessage :: (EsqDBFlow m r, CacheFlow m r) => Id DM.Merchant -> DMM.MessageKey -> BuildGenericMessageReq -> m Text
+buildGenericMessage merchantId messageKey _ = do
+  merchantMessage <-
+    QMM.findByMerchantIdAndMessageKey merchantId messageKey
```

Comment:
is it cached, if not then can we run this in replica ?

Reply:
`import qualified Storage.CachedQueries.Merchant.MerchantMessage as QMM`

( It is a cached query. In this file, the qualified import is named as `QMM`